### PR TITLE
Move jedi CI container spec for bufr from version 12.0.0 to 12.0.1

### LINKED ISSUE #867: release/1.5.1 CI containers don't have bufr@12.0.1, but bufr@12.0.0
**Describe the bug**
When `bufr@12.0.1` was added for release 1.5.1, the container specs didn't get updated and therefore the 1.5.1 JEDI CI containers still have `bufr@12.0.0`.

**To Reproduce**
See https://github.com/JCSDA/spack-stack/blob/883ebcb2509df68702a7207bdd410926f20e3e9c/configs/containers/README.md?plain=1#L8C58-L8C58

**Expected behavior**
We wanted 12.0.1, not 12.0.0.

**System:**
Docker containers used by JEDI CI

**Additional context**
Since the next release 1.6.0 will be made before Christmas, there is no point going back and fixing it in the release. Instead, update https://github.com/JCSDA/spack-stack/blob/651ff12482b6f583c8e90a58ec826ec61947c9d2/configs/containers/specs/jedi-ci.yaml#L3 after https://github.com/JCSDA/spack-stack/pull/866 was merged.

### PR DESCRIPTION
**Pull request by srherbener**
### Summary

This PR moves the version for bufr from 12.0.0 to 12.0.1 in the jedi ci container spec. This is needed to be able to test ioda-converters properly with the latest bufr2ioda.x application.

### Testing

Only the CI testing.

### Applications affected

JEDI CI testing

### Systems affected

JEDI CI test systems (AWS)

### Dependencies

None

### Issue(s) addressed

Resolves #867 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by srherbener** [comment]
I'm making this a draft PR because I want to test it, but it needs to wait until after #866 is merged.

**Comment by climbfuji** [comment]
CI will only test this nightly (after being merged) or via workflow dispatch, but that only makes sense after #866 is merged. The change is correct, thanks for that.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
